### PR TITLE
Flutter auto-extract assets

### PR DESCRIPTION
**Pull request by laves**
Flutter package can now extract Rhino context files and model files automatically.
- binding changed to support this
- demo changed to demonstrate the change
- tested on Android and iOS

### CODE REVIEW

File: binding/flutter/README.md [hunk 1]
```diff
@@ -201,26 +201,13 @@ flutter:
...
-
-    ByteData data = await rootBundle.load(resourcePath);
-    final buffer = data.buffer;
-
-    await outputFile.create(recursive: true);
-    await outputFile.writeAsBytes(
-        buffer.asUint8List(data.offsetInBytes, data.lengthInBytes));
-    return outputPath;
+try{
+    _rhino = await Rhino.create(contextAsset);
```
Comment (dbartle) [inline binding/flutter/README.md:208]: I think the README diff here shows a real win in terms of usability

File: demo/flutter/lib/main.dart [hunk 2]
```diff
@@ -48,7 +44,9 @@ class _MyAppState extends State<MyApp> {
   }
 
   Future<void> initRhino() async {
-    String contextPath = await _extractDemoContext();
+    String platform = Platform.isAndroid ? "android" : "ios";
```
Comment (dbartle) [inline demo/flutter/lib/main.dart:47]: This is pedantic, but explicitly check for "isIos"

I've seen this type of logic blow up in weird ways before when something that was originally 2 choices increases to 3, or some other underlying assumption changes. We can simply throw if it is a thirdt ype for now with and then output the $platform in the error string.

File: demo/flutter/lib/main.dart [hunk 3]
```diff
@@ -92,25 +90,6 @@ class _MyAppState extends State<MyApp> {
     return printText;
   }
 
-  Future<String> _extractDemoContext() async {
```
Comment (dbartle) [inline demo/flutter/lib/main.dart:95]: stoked to see this

Comment (laves) [reply demo/flutter/lib/main.dart:95]: I was very stoked to delete it!

**Review by dbartle — APPROVED** [review]

**Pull request merged**